projects/nginx/conf_fuzzer.cc

```diff
@@ -0,0 +1,53 @@
+// Copyright 2016 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include <stdlib.h>
+#include <string.h>
+#include <iostream>
+#include <fstream>
+
+#include <unistd.h>
+#include <sys/types.h>
+#include <sys/wait.h>
+
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+  std::string out_path, conf_path, cmd;
+  std::ofstream fc;
+  int wstatus;
+  const char *env[] = {"ASAN_OPTIONS=detect_leaks=0", NULL};
+
+  out_path = getenv("OUT");
+  conf_path = out_path + "/conf/nginx.conf";
+
+  fc.open(conf_path, std::ios::out | std::ios::binary);
+  fc.write((char*) data, size);
+  fc.close();
+
+  pid_t pid = fork();
```

Comment:
+1, I believe there is a way to test config without creating a file on the disk and starting another process. Consider the following routine:

1) use `fmemopen` to create a stream in memory: http://man7.org/linux/man-pages/man3/fmemopen.3.html
2) use `fileno` to get a file descriptor from the stream pointer: http://man7.org/linux/man-pages/man3/fileno.3p.html
3) write `data` to that in-memory-file, it may require calling `fflush` after `write`
4) initialize `ngx_conf_t` object with that descriptor in a proper way: similar to https://github.com/nginx/nginx/blob/master/src/core/ngx_conf_file.c#L176
5) call `ngx_conf_parse` without `filename` argument, as `*cf` already has been initialized and is ready for parsing
6) close the file in the end :)

Probably I could miss something, as I'm not familiar with nginx source code, but that approach should work in general.